src/libmongoc/src/mongoc/mongoc-client-session.c

```diff
@@ -417,6 +424,11 @@ mongoc_session_opts_set_causal_consistency (mongoc_session_opt_t *opts,
    BSON_ASSERT (opts);
 
    if (causal_consistency) {
+      if (mongoc_session_opts_get_snapshot (opts)) {
+         MONGOC_WARNING (
+            "Cannot set causal consistency when session is snapshot");
+         EXIT;
```

Comment:
I'm _sure_ this is not the right way to "error". Any ideas?

Reply:
Perfect. I added an error with `MONGOC_ERROR_CLIENT` and `MONGOC_ERROR_CLIENT_SESSION_FAILURE` in `mongoc_client_start_session` instead.